diffusion/datasets/image_caption.py

```diff
@@ -0,0 +1,208 @@
+        tokenizer_name_or_path (str): The name or path of the tokenizer to use. Default: ``'stabilityai/stable-diffusion-2-base'``.
+        caption_drop_prob (float): The probability of dropping a caption. Default: ``0.0``.
+        resize_size (int): The size to resize the image to. Default: ``256``.
+        num_samples (int, optional): The number of samples to use. Default: ``None`` uses all available samples.
+        predownload (int, optional): The number of samples to prefetch. Default: ``100_000``.
+        download_retry (int): The number of times to retry a download. Default: ``2``.
+        download_timeout (float): The timeout for a download. Default: ``120``.
+        drop_last (bool): Whether to drop the last batch if it is incomplete. Default: ``True``.
+        shuffle (bool): Whether to shuffle the samples in this dataset. Default: ``True``.
+        num_canonical_nodes (int, optional): The number of canonical nodes for shuffle. Default: ``None``.
+        caption_selection (str): If there are multiple captions, specifies how to select a single caption.
+            'first' selects the first caption in the list and 'random' selects a random caption in the list.
+            If there is only one caption, this argument is ignored. Default: ``'first'``.
+        transform (Optional[Callable]): The transforms to apply to the image. Default: ``None``.
+        image_key (str): Key associated with the image in the streaming dataset. Default: ``'image'``.
+        caption_key (str): Key associated with the caption in the streaming dataset. Default: ``'caption'``.
+        is_precomputed_latents (bool): Whether or not to use the precomputed latents streaming dataset.
+            Default: ``False``
+        **dataloader_kwargs: Additional arguments to pass to the dataloader.
+    """
+    if isinstance(remote, str) and isinstance(local, str):
+        # Hacky... make remote and local lists to simplify downstream code
+        remote, local = [remote], [local]
+    elif isinstance(remote, Sequence) and isinstance(local, Sequence):
+        if len(remote) != len(local):
+            ValueError(
+                f'remote and local Sequences must be the same length, got lengths {len(remote)} and {len(local)}')
+    else:
+        ValueError(f'remote and local must be both Strings or Sequences, got types {type(remote)} and {type(local)}.')
+
+    # Create a Stream for each (remote, local) pair
+    streams = []
+    for r, l in zip(remote, local):
+        streams.append(Stream(remote=r, local=l, download_retry=download_retry, download_timeout=download_timeout))
+
+    # Setup the transforms to apply
+    if transform is None:
+        transform = [
+            LargestCenterSquare(resize_size),
+            transforms.ToTensor(),
+            transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),  # # Normalize from 0 to 1 to -1 to 1
+        ]
+    transform = transforms.Compose(transform)
+
+    dataset = StreamingImageCaptionDataset(
```

Comment:
We should add an option to set `cache_limit` for the dataset.

Reply:
Should be covered by the new `streaming_kwargs`